test/jdk/sun/tools/jstatd/JstatGCUtilParser.java

```diff
@@ -93,21 +93,23 @@ public static void verify(String... valueArray) throws Exception {
             for (int i = 0; i < values().length; i++) {
                 GcStatisticsType type = values()[i].getType();
                 String value = valueArray[i].trim();
+                if ((type.equals(GcStatisticsType.PERCENTAGE_OR_DASH)
+                     || type.equals(GcStatisticsType.INTEGER_OR_DASH)
+                     || type.equals(GcStatisticsType.DOUBLE_OR_DASH))
+                    && value.equals("-")) {
+                    continue;
+                }
                 if (type.equals(GcStatisticsType.INTEGER)) {
                     NumberFormat.getInstance().parse(value).intValue();
-                    break;
+                    continue;
                 }
                 if (type.equals(GcStatisticsType.DOUBLE)) {
                     NumberFormat.getInstance().parse(value).doubleValue();
-                    break;
-                }
-                if (type.equals(GcStatisticsType.PERCENTAGE_OR_DASH) &&
-                        value.equals("-")) {
-                    break;
+                    continue;
                 }
                 double percentage = NumberFormat.getInstance().parse(value).doubleValue();
                 assertTrue(0 <= percentage && percentage <= 100,
-                        "Not a percentage: " + value);
+                        "Not a percentage. value: " + value + " percentage: " + percentage);
```

Comment:
This doesn't look right. If the type is `INTEGER_OR_DASH` and the value is not a "-", then we don't do the `NumberFormat` check that we previously would have done if the type was `INTEGER`. Instead we treat it as a double and assert that the value is between 0 and 100. Same issue with `DOUBLE_OR_DASH`.